src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int capacity;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        this.threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        this.capacity = DEFAULT_CAPACITY;
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int insertIndex = findElementIndex(key);
+        Node<K, V> node = table[insertIndex];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        table[insertIndex] = new Node<>(key, value, table[insertIndex]);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int nodeIndex = findElementIndex(key);
+        Node<K, V> node = table[nodeIndex];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int findElementIndex(K key) {
```

Comment:
Private methods should be placed after all public methods in the class. Consider moving this method below the `resize` method.